include/ccf/ds/attestation_types.h

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#if defined(INSIDE_ENCLAVE) && !defined(VIRTUAL_ENCLAVE)
+#  include <openenclave/attestation/attester.h>
+#  include <openenclave/attestation/custom_claims.h>
+#  include <openenclave/attestation/sgx/evidence.h>
+#  include <openenclave/attestation/verifier.h>
+
+#endif
+
+#include <array>
+
+namespace ccf
+{
+  static constexpr size_t attestation_report_data_size = 32;
+  using attestation_report_data =
+    std::array<uint8_t, attestation_report_data_size>;
+
+#if defined(INSIDE_ENCLAVE) && !defined(VIRTUAL_ENCLAVE)
+  static constexpr size_t attestation_measurement_size = 32;
+  using attestation_measurement =
+    std::array<uint8_t, attestation_measurement_size>;
+  // Set of wrappers for safe memory management
+  struct Claims
+  {
+    oe_claim_t* data = nullptr;
+    size_t length = 0;
+
+    ~Claims()
+    {
+      oe_free_claims(data, length);
+    }
+  };
+
+  struct CustomClaims
+  {
+    oe_claim_t* data = nullptr;
+    size_t length = 0;
+
+    ~CustomClaims()
+    {
+      oe_free_custom_claims(data, length);
+    }
+  };
+
+  struct SerialisedClaims
+  {
+    uint8_t* buffer = nullptr;
+    size_t size = 0;
+
+    ~SerialisedClaims()
+    {
+      oe_free_serialized_custom_claims(buffer);
+    }
+  };
+
+  struct Evidence
+  {
+    uint8_t* buffer = NULL;
+    size_t size = 0;
+
+    ~Evidence()
+    {
+      oe_free_evidence(buffer);
+    }
+  };
+
+  struct Endorsements
+  {
+    uint8_t* buffer = NULL;
+    size_t size = 0;
+
+    ~Endorsements()
+    {
+      oe_free_endorsements(buffer);
+    }
+  };
+
+  static constexpr oe_uuid_t oe_quote_format = {OE_FORMAT_UUID_SGX_ECDSA};
+  static constexpr auto sgx_report_data_claim_name = OE_CLAIM_SGX_REPORT_DATA;
+
+#else
+  static constexpr size_t attestation_measurement_size = 48;
+  using attestation_measurement =
+    std::array<uint8_t, attestation_measurement_size>;
+
+  union tcb_version {
```

Comment:
It would be nice to move the AMD-specific types to a new file.

Reply:
Done a full split into attestation_sgx.h and attestation_sev_snp.h